src/inmanta/agent/agent.py

```diff
@@ -285,7 +295,7 @@ def execute(self, dummy, generation, cache):
                              result.result)
 
             status = const.ResourceState[result.result["resource"]["status"]]
-            if status == const.ResourceState.available or self.future.done():
+            if status == const.ResourceState.available or status == const.ResourceState.deploying or self.future.done():
```

Comment:
I would prefer something like status in const.TRANSIENT_STATES similar to UNDEPLOYABLE_STATES now. Otherwise this error is bound to happen again.